### ISSUE TITLE
Expand producing orphan edges for option group query

### ISSUE THREAD

**Issue by amykglen** [RTXteam/RTX#1513]
query reported by @dkoslicki:
```
add_qnode(ids=CHEMBL.COMPOUND:CHEMBL1200633, key=n0)
# Non-optional nodes and edges
add_qnode(key=n11)
add_qnode(key=n3,ids=MONDO:0100096)
add_qedge(subject=n0, object=n11, key=e00)
add_qedge(subject=n11, object=n3, key=e11)
# optional nodes and edges
add_qnode(key=n1,option_group_id=G1)
add_qnode(key=n2 ,option_group_id=G1)
add_qedge(subject=n0, object=n1, key=e0 ,option_group_id=G1)
add_qedge(subject=n1, object=n2, key=e1 ,option_group_id=G1)
add_qedge(subject=n2, object=n3, key=e2 ,option_group_id=G1)
# then expand, get results, and filter (no NGD for speed purposes)
expand()
resultify()
filter_results(action=limit_number_of_results, max_results=50)
```
upon some preliminary digging: looks like expand isn't finding answers for edge e1, and then pruning isn't happening properly after that. and it seems to be that the e1 query is isn't finding any answers because it's timing out (it's huge - asking for `(3577 curies)--[any predicate]--(any category)`). so seems to be two issues here.

**Comment by amykglen** [comment]
attaching a file containing the e1 query that's sent to the KG2 API (too big for pasting in):
[e1_query_for_kg2_api.txt](https://github.com/RTXteam/RTX/files/6659234/e1_query_for_kg2_api.txt)

**Comment by saramsey** [comment]
Is this sort of query the kind of thing we would just use Pathfinder for today, @dkoslicki ?  Wondering if we can close this issue.

**Comment by dkoslicki** [comment]
Yes, this does appear to be a pathfinder-esque query (looking for 2 or 3-hop paths between two pinned nodes). But that doesn't address the orphaned edge issue. Given that the UI isn't really using optional edges, I think we're ok to close
